debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleSnapshotChangeEventSource.java

```diff
@@ -186,6 +186,15 @@ private boolean areSameTimestamp(Long scn1, long scn2) throws SQLException {
 
             return Optional.of(rs.getLong(1));
         }
+        catch (SQLException e) {
+            if (e.getMessage().contains("no snapshot found based on specified time")) {
```

Comment:
Wouldn't it better to match against the ORA error code (i.e ORA-08180)? 
Asking as well whether there is no a better/programmatically manner to check for this cases instead of relying on error string matching.

Reply:
We can definitely match it against `startsWith("ORA-08180")`.

Reply:
@Naros As we match against the error code now I updated to code to use error code from the exception isnated of message